src/main/java/com/iota/iri/Iota.java

```diff
@@ -40,90 +39,48 @@
-            if(StringUtils.isNotEmpty(coordinatorTrytes)) {
-                coordinator = new Hash(coordinatorTrytes);
-            } else {
-                log.warn("No coordinator address given for testnet. Defaulting to "
-                        + Configuration.TESTNET_COORDINATOR_ADDRESS);
-                coordinator = new Hash(Configuration.TESTNET_COORDINATOR_ADDRESS);
-            }
-        } else {
-            coordinator = new Hash(Configuration.MAINNET_COORDINATOR_ADDRESS);
-        }
+        Snapshot initialSnapshot = Snapshot.init(configuration.getSnapshotFile(),
+                configuration.getSnapshotSignatureFile(), configuration.isTestnet()).clone();
         tangle = new Tangle();
-        messageQ = new MessageQ(configuration.integer(Configuration.DefaultConfSettings.ZMQ_PORT),
-                configuration.string(Configuration.DefaultConfSettings.ZMQ_IPC),
-                configuration.integer(Configuration.DefaultConfSettings.ZMQ_THREADS),
-                configuration.booling(Configuration.DefaultConfSettings.ZMQ_ENABLED)
-                );
+        messageQ = MessageQ.createWith(configuration);
         tipsViewModel = new TipsViewModel();
         transactionRequester = new TransactionRequester(tangle, messageQ);
         transactionValidator = new TransactionValidator(tangle, tipsViewModel, transactionRequester, messageQ,
-                snapshotTimestamp);
-        milestone = new Milestone(tangle, coordinator, initialSnapshot, transactionValidator, testnet, messageQ,
-                numKeysMilestone, milestoneStartIndex, dontValidateMilestoneSig);
-        node = new Node(configuration, tangle, transactionValidator, transactionRequester, tipsViewModel, milestone, messageQ);
-        replicator = new Replicator(node, tcpPort, maxPeers, testnet, transactionPacketSize);
-        udpReceiver = new UDPReceiver(udpPort, node, configuration.integer(Configuration.DefaultConfSettings.TRANSACTION_PACKET_SIZE));
+                configuration);
+        milestone = new Milestone(tangle, transactionValidator, messageQ, configuration, initialSnapshot);
+        node = new Node(configuration, tangle, transactionValidator, transactionRequester, tipsViewModel, milestone,
+                messageQ);
+        replicator = new Replicator(node, configuration);
+        udpReceiver = new UDPReceiver(node, configuration);
         ledgerValidator = new LedgerValidator(tangle, milestone, transactionRequester, messageQ);
         tipsSolidifier = new TipsSolidifier(tangle, transactionValidator, tipsViewModel);
-        tipsSelector = createTipSelector(milestoneStartIndex, alpha, belowMaxDepthTxLimit);
+        tipsSelector = createTipSelector(configuration);
     }
 
     public void init() throws Exception {
         initializeTangle();
         tangle.init();
 
-        if (configuration.booling(Configuration.DefaultConfSettings.RESCAN_DB)){
+        if (configuration.isRescanDb()){
```

Comment:
shouldn't these be called `shouldRescanDb` and `shouldRevalidate`?

Reply:
Even apparently `should` seems to be an approved prefix by at least some people (https://wiki.eclipse.org/Recommenders/CodingConventions#is_prefix_should_be_used_for_boolean_variables_and_methods.), it is not part of a Java Bean specification (https://docstore.mik.ua/orelly/java-ent/jnut/ch06_02.htm).

Why do I care? We use Jackson to deserialize the INI file directly to the Config object. Jackson relies on Java Beans conventions. So the INI parsing may break by this change.

I can do another thing. I can put `@jsonProperty("rescanDb")` annotation above the getter.  This will free the code from following a certain convention.

I chose not to do this because it adds verbosity to the code...
However, after your comment here I am now fearing that I am actually obscuring the fact that this is class is being deserialized by Jackson and someone may incidentally perform a change that will break this.

I suppose that if I add the annotation to one getter, then I should add them to all.
I will add a comment to the `NetConfig` class that explains the matter and open an issue to use Jackson annotations.